### PR TITLE
Closes #3743: Adding TrackingProtectionPolicy.CookiePolicy.

### LINKED ISSUE #3743: Add API for indicating how cookies should behave for a given TrackingProtectionPolicy
Fenix related issue: https://github.com/mozilla-mobile/fenix/issues/1749

### PR DESCRIPTION
**Pull request by Amejia481**
As GeckoView is [enabling by default  ETP(Enhanced Tracking Protection)](https://github.com/mozilla-mobile/fenix/issues/1749#issuecomment-509517270) by default we are following the same steps.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt [hunk 1]
```diff
@@ -209,11 +211,35 @@ class GeckoEngineTest {
...
+                cookiePolicy = CookiePolicy.ACCEPT_ONLY_FIRST_PARTY
+            )
+
+        assertEquals(CookiePolicy.ACCEPT_ONLY_FIRST_PARTY.id, contentBlockingSettings.cookieBehavior)
+
+        engine.settings.trackingProtectionPolicy = TrackingProtectionPolicy.none()
+
+        assertEquals(CookiePolicy.ACCEPT_ALL.id, contentBlockingSettings.cookieBehavior)
+
+        engine.settings.trackingProtectionPolicy =
```
Comment (csadilek) [inline components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt]: This assertion seems redundant? Isn't it the same as line 221+. You select a TP policy with a cookiePolicy and assert that the cookie policy is set correctly. Seems to me both blocks test the same?

File: components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt [hunk 2]
```diff
@@ -209,11 +211,33 @@ class GeckoEngineTest {
...
+        engine.settings.trackingProtectionPolicy =
+            TrackingProtectionPolicy.select(TrackingProtectionPolicy.AD, cookiePolicy = CookiePolicy.ACCEPT_ONLY_FIRST_PARTY)
+
+        assertEquals(CookiePolicy.ACCEPT_ONLY_FIRST_PARTY.id, contentBlockingSettings.cookieBehavior)
+
+        engine.settings.trackingProtectionPolicy = TrackingProtectionPolicy.none()
+
+        assertEquals(CookiePolicy.ACCEPT_ALL.id, contentBlockingSettings.cookieBehavior)
+
+        engine.settings.trackingProtectionPolicy =
```
Comment (csadilek) [inline components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt]: nit: same comment as above. This seems to test the same as line 222+

File: components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt [hunk 3]
```diff
@@ -155,8 +157,35 @@ class GeckoEngineTest {
...
+        assertEquals(
+            TrackingProtectionPolicy.CookiePolicy.ACCEPT_ONLY_FIRST_PARTY.id,
+            contentBlockingSettings.cookieBehavior
+        )
+
+        engine.settings.trackingProtectionPolicy = TrackingProtectionPolicy.none()
+
+        assertEquals(TrackingProtectionPolicy.CookiePolicy.ACCEPT_ALL.id, contentBlockingSettings.cookieBehavior)
+
+        engine.settings.trackingProtectionPolicy =
```
Comment (csadilek) [inline components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt]: nit: same as above...

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt [hunk 4]
```diff
@@ -143,27 +183,30 @@ abstract class EngineSession(
...
-            internal const val ALL = RECOMMENDED + CRYPTOMINING + FINGERPRINTING + CONTENT
+            const val ALL = RECOMMENDED + CRYPTOMINING + FINGERPRINTING + CONTENT
 
-            fun none() = TrackingProtectionPolicy(NONE)
+            fun none() = TrackingProtectionPolicy(NONE, cookiePolicy = ACCEPT_ALL)
 
             /**
              * Strict policy.
              * Combining the [recommended] categories plus [CRYPTOMINING], [FINGERPRINTING] and [CONTENT].
+             *  With a cookiePolicy of [ACCEPT_NON_TRACKERS].
```
Comment (csadilek) [inline components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt]: nit: unneeded space.

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt [hunk 5]
```diff
@@ -143,27 +183,30 @@ abstract class EngineSession(
...
             /**
              * Recommended policy.
              * Combining the [AD], [ANALYTICS], [SOCIAL], [TEST] categories plus [SAFE_BROWSING_ALL].
+             * With a [CookiePolicy] of [ACCEPT_NON_TRACKERS].
              * This is the recommended setting.
              */
             fun recommended() = TrackingProtectionPolicyForSessionTypes(RECOMMENDED)
 
-            fun select(vararg categories: Int) = TrackingProtectionPolicyForSessionTypes(categories.sum())
+            fun select(vararg categories: Int, cookiePolicy: CookiePolicy = ACCEPT_NON_TRACKERS) =
```
Comment (csadilek) [inline components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt]: Oh did you decide not to go with a fluent API? I think this is fine / simple enough for now.

File: docs/changelog.md [hunk 6]
```diff
@@ -28,6 +28,11 @@ permalink: /changelog/
   * See component's [README](https://github.com/mozilla-mobile/android-components/blob/master/components/service/firefox-accounts/README.md) for detailed description of the new API.
   * As part of these changes, token caching issue has been fixed. See [#3579](https://github.com/mozilla-mobile/android-components/pull/3579) for details.
 
+* **concept-engine**, **browser-engine-gecko(-beta/nightly)**.
+  * Added `TrackingProtectionPolicy.CookiePolicy()` to indicate how cookies should behave for a given `TrackingProtectionPolicy`.
```
Comment (csadilek) [inline docs/changelog.md]: nit `TrackingProtectionPolicy.CookiePolicy` no ().

File: docs/changelog.md [hunk 7]
```diff
@@ -28,6 +28,11 @@ permalink: /changelog/
...
+  * Now `TrackingProtectionPolicy.select` allows you to indicate a `TrackingProtectionPolicy.CookiePolicy()`, if you don't specified any the `TrackingProtectionPolicy.CookiePolicy.ACCEPT_NON_TRACKERS` will be assigned.
```
Comment (csadilek) [inline docs/changelog.md]: allows you to specify ... ? If not specified, `CookiePolicy.ACCEPT_NON_TRACKERS` will be used.

File: docs/changelog.md [hunk 8]
```diff
@@ -28,6 +28,11 @@ permalink: /changelog/
...
+  * Behavior change: Now `TrackingProtectionPolicy.none()` will get assigned a `TrackingProtectionPolicy.CookiePolicy.ACCEPT_ALL`, and both `TrackingProtectionPolicy.all()` and `TrackingProtectionPolicy.recommended()`  will have a `TrackingProtectionPolicy.CookiePolicy.ACCEPT_NON_TRACKERS`.
```
Comment (csadilek) [inline docs/changelog.md]: 👍

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt (hunk 5 above)
Comment (Amejia481) [reply components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt]: Yes, I was playing with the fluent API that you suggested me, and I realized that consumers only care about cookie policies, when they want something different from the defaults, this solution was simple enough to allow them customization without introducing a new API.

File: components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt (hunk 1 above)
Comment (Amejia481) [reply components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt]: Fixed, thanks!

File: components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt (hunk 3 above)
Comment (Amejia481) [reply components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt]: Fixed!

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt (hunk 4 above)
Comment (Amejia481) [reply components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt]: Fixed!

File: docs/changelog.md (hunk 6 above)
Comment (Amejia481) [reply docs/changelog.md]: Fixed!

File: docs/changelog.md (hunk 7 above)
Comment (Amejia481) [reply docs/changelog.md]: Fixed!

File: components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt (hunk 2 above)
Comment (Amejia481) [reply components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt]: Fixed!

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt (hunk 5 above)
Comment (csadilek) [reply components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt]: ok, makes sense!

**Pull request merged**